torchrl/envs/gym_like.py

```diff
@@ -133,18 +133,24 @@ def read_action(self, action):
         """
         return self.action_spec.to_numpy(action, safe=False)
 
-    def read_done(self, done):
+    def read_done(self, done, termination, truncation):
         """Done state reader.
 
-        Reads a done state and returns a tuple containing:
+        In TorchRL, done means that a trajectory is terminated (we do not support the
+        terminated signal). Truncated means the trajectory has been interrupted.
+
+        This method returns:
         - a done state to be set in the environment
         - a boolean value indicating whether the frame_skip loop should be broken
 
         Args:
             done (np.ndarray, boolean or other format): done state obtained from the environment
-
+            termination (bool or None): termination signal.
+            truncation (bool or None): truncation signal.
         """
-        return done, done
+        if termination is not None:
+            return termination, truncation, done
+        return done, truncation, done
```

Comment:
Let me rephrase what you're saying here:
if older version of gym (<0.26) a truncation signal could be retrieved from the info. 
Currently, we don't use it so assuming that done=truncated will not work properly with value functions, is that right?

If so, I think that we should advise users of adopting a dedicated info reader to write the truncated signal, which will solve the problem.

Reply:
> if older version of gym (<0.26) a truncation signal could be retrieved from the info.

Yes in the `TimeLimit.truncated` key if it happened. See https://gymnasium.farama.org/tutorials/gymnasium_basics/handling_time_limits/#solution.

> assuming that done=truncated will not work properly with value functions

Yes, and to be exact done here with the old gym versions would be `gym_done=terminated | truncated`. (It was an `end_of_traj`)

> I think that we should advise users of adopting a dedicated info reader to write the truncated signal, which will solve the problem.

Yes, that would be the correct way to record the `truncated` key and propagate it to the rest of TorchRL's components. However, the  `gym_done` key would still need to be corrected with `torchrl_done = gym_done and ~truncated` I don't how this could be done by the users.